azure/packages/test/scenario-runner/src/DocLoaderRunner.ts

```diff
@@ -4,31 +4,58 @@
  */
 import child_process from "child_process";
 
+import { ConnectionState } from "fluid-framework";
+import { AzureClient } from "@fluidframework/azure-client";
 import { TypedEventEmitter } from "@fluidframework/common-utils";
+import { IFluidContainer } from "@fluidframework/fluid-static";
+import { PerformanceEvent } from "@fluidframework/telemetry-utils";
+import { timeoutPromise } from "@fluidframework/test-utils";
 
-import { IRunConfig, IRunner, IRunnerEvents, IRunnerStatus, RunnnerStatus } from "./interface";
-import { delay } from "./utils";
+import {
+	AzureClientConnectionConfig,
+	ContainerFactorySchema,
+	IRunConfig,
+	IRunner,
+	IRunnerEvents,
+	IRunnerStatus,
+	RunnnerStatus,
+} from "./interface";
+import {
+	AzureClientConfig,
+	createAzureClient,
+	delay,
+	getScenarioRunnerTelemetryEventMap,
+	loadInitialObjSchema,
+} from "./utils";
+import { getLogger } from "./logger";
 
-export interface AzureClientConfig {
-	type: "remote" | "local";
-	endpoint?: string;
-	key?: string;
-	tenantId?: string;
-	useSecureTokenProvider?: boolean;
-	region?: string;
-}
-
-export interface DocLoaderSchema {
-	initialObjects: { [key: string]: string };
-	dynamicObjects?: { [key: string]: string };
-}
+const eventMap = getScenarioRunnerTelemetryEventMap("DocLoader");
 
 export interface DocLoaderRunnerConfig {
-	connectionConfig: AzureClientConfig;
-	schema: DocLoaderSchema;
+	connectionConfig: AzureClientConnectionConfig;
+	schema: ContainerFactorySchema;
 	docIds: string[];
 	clientStartDelayMs: number;
 	numOfLoads?: number;
+	client?: AzureClient;
+}
+
+export interface DocLoaderRunnerRunConfig
```

Comment:
Hmm - maybe a name could be better. Don't have immediate idea but anything different then *RunnerRun*. Its also quite similar to DocLoaderRunnerConfig.

Reply:
Agreed.